src/odemis/driver/hamamatsurx.py

```diff
@@ -817,64 +895,61 @@ def __init__(self, name, role, port, host, children=None, daemon=None, **kwargs)
             self._readoutcam = ReadoutCamera(parent=self, spectrograph=children.get("spectrograph"),
                                              daemon=daemon, **kwargs)
             self.children.value.add(self._readoutcam)  # add readoutcam to children-VA
             try:
                 kwargs = children["streakunit"]
             except Exception:
-                raise
+                raise ValueError("Required child streakunit not provided")
             self._streakunit = StreakUnit(parent=self, daemon=daemon, **kwargs)
             self.children.value.add(self._streakunit)  # add streakunit to children-VA
             try:
                 kwargs = children["delaybox"]
             except Exception:
-                raise
+                raise ValueError("Required child delaybox not provided")
             self._delaybox = DelayGenerator(parent=self, daemon=daemon, **kwargs)
             self.children.value.add(self._delaybox)  # add delaybox to children-VA
 
     def _openConnection(self):
         """
-        open connection with RemoteEx client.
-        :parameter host: IP-adress or hostname
-        :parameter port: port for sending/receiving commands
-        :parameter port_d: port for reading images
-        return: connection to RemoteEx command and data port
+        Open connection with RemoteEx client.
+        :return: connection to RemoteEx command and data port
         """
         # connect to sockets
         try:
             self._commandport = socket.create_connection((self.host, self.port), timeout=5)
             self._dataport = socket.create_connection((self.host, self.port_d), timeout=5)
-        except socket.timeout as msg:
-            raise model.HwError(msg, "Failed to connect to '%s using port %d'. Check the server "
+        except (socket.timeout, socket.error):
+            raise model.HwError("Failed to connect to host %s using port %d'. Check the server "
                                 "is connected to the network, turned "
                                 " on, and correctly configured." % (self.host, self.port))
-        except socket.error as msg:
-            raise model.HwError(msg, "Failed to connect to '%s:%d'. Check ...." % (self.host, self.port))
 
         # check if connection returns correct response
         try:
             message = self._commandport.recv(self.port)
             if message != 'RemoteEx Ready\r':
-                raise ValueError("Connection to port %s not successfull. "
+                raise ValueError("Connection to port %s not successful. "
                                  "Response %s from server is not as expected." % (self.port, message))
-        except socket.timeout as msg:
-            raise model.HwError(msg, "Failed to receive response from '%s:%d'. Check ..." % (self.host, self.port))
+        except socket.timeout:
+            raise model.HwError("Failed to receive response from %s:%d that commandport"
```

Comment:
A space is missing between "commandport" and "is". Anyway, keep in mind that the message in HwError will be displayed to the user when initialisation fails. So try to be less technical, and give some advice on how to solve the issue. Maybe something like: "Hamamastu RemoteEx didn't respond. Check that it is running properly, or restart the Streak Camera computer".